test/test_nn.py

```diff
@@ -13390,18 +13390,28 @@ def test_conv3d_valid_padding_backward(self, device):
         # NnpackSpatial, Winograd3x3Depthwise, and Xnnpack2d backends. Testing these
         # requires the ability to gate tests by whether PyTorch is built with USE_MOBILE=1.
     ])
+    # Test with both stride=1 and stride>1 cases.
     @parametrize_test("strided", [False, True])
-    def test_conv_backend_selection(
-            self, device, input_shape, strided, transposed, dilated, groups, layout, backend_expected):
+    # Test with both contiguous and non-contiguous inputs.
+    @parametrize_test("contiguous", [False, True])
+    def test_conv_backend(
+            self, device, input_shape, strided, contiguous, transposed, dilated, groups, layout, backend_expected):
         # Build up inputs.
         dtype = torch.float32
         C_in, C_out, dim, kernel_size = input_shape[1], 6, len(input_shape) - 2, 3
-        x = torch.empty(*input_shape, device=device, dtype=dtype, layout=layout, requires_grad=True)
-        weight = torch.empty(C_in if transposed else C_out,
+        x = torch.randn(*input_shape, device=device, dtype=dtype, requires_grad=True)
+        if not contiguous:
+            x = torch.repeat_interleave(x, 2, dim=-1)
+            x = x[..., ::2].detach()
```

Comment:
We should also probably test non-contiguous weight and non-contiguous bias

Reply:
Yep good catch!